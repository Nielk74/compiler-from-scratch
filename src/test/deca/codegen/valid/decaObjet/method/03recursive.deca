// Description:
//      Recursive function calls.
//
// Resultats:
//    02recursive.deca
//
// Historique:
//    cree le 16/06/2022


class A {

    int factorial(int n){
        
        if (n == 0){
            return 1;    
        } 

        else  {
            return  n * factorial(n-1) ;
        }  
    }       
}

{
    A a = new A();
    println(a.factorial(0));
    println(a.factorial(4));
    println(a.factorial(10));
}