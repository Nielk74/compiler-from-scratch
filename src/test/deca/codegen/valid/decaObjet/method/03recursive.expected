1
24
3628800
